raster/src/main/scala/geotrellis/raster/render/ColorMap.scala

```diff
@@ -204,85 +206,60 @@ class IntColorMap(breaksToColors: Map[Int, Int], val options: Options = Options.
-        case BTree(v, l, _)       if z < v._1                       => Left(l)
-        case BTree(_, _, r)                                         => Left(r)
-      }}
-      case LessThanOrEqualTo => { (z, tree) => tree match {
-        case BTree(v, None, _)    if z <= v._1                      => Right(v)
-        case BTree(v, Some(l), _) if z <= v._1 && z > l.greatest._1 => Right(v)
-        case BTree(v, l, _)       if z < v._1                       => Left(l)
-        case BTree(_, _, r)                                         => Left(r)
-      }}
-      case Exact => { (z, tree) => tree match { /* Vanilla Binary Search */
-        case BTree(v, _, _) if z == v._1 => Right(v)
-        case BTree(v, l, _) if z < v._1  => Left(l)
-        case BTree(_, _, r)              => Left(r)
-      }}
-      case GreaterThanOrEqualTo => { (z, tree) => tree match {
-        case BTree(v, _, None)    if z >= v._1                    => Right(v)
-        case BTree(v, _, Some(r)) if z >= v._1 && z < r.lowest._1 => Right(v)
-        case BTree(v, l, _)       if z < v._1                     => Left(l)
-        case BTree(_, _, r)                                       => Left(r)
-      }}
-      case GreaterThan => { (z, tree) => tree match {
-        case BTree(v, _, None)    if z > v._1                     => Right(v)
-        case BTree(v, _, Some(r)) if z > v._1 && z <= r.lowest._1 => Right(v)
-        case BTree(v, l, _)       if z <= v._1                    => Left(l) /* (<=) is correct here! */
-        case BTree(_, _, r)                                       => Left(r)
-      }}
-    }
-  }
-
-  /* Horrible assumption: `breaksToColors` isn't empty */
-  private lazy val colourTree: BTree[(Int, Int)] =
-    BTree.fromSortedSeq(breaksToColors.toIndexedSeq.sorted).get
-
-  def map(z: Int): Int = {
-    if(isNoData(z)) {
-      options.noDataColor
-    } else {
-      colourTree.searchWith(z, branchPred) match {
-        case Some((_, colour)) => colour
-        case None if options.strict => sys.error(s"Value $z did not have an associated color and break")
-        case _ => options.fallbackColor
-      }
-    }
-  }
+  def map(z: Int): Int = breakMap(z)
 
-  def mapDouble(z: Double): Int =
-    map(d2i(z))
+  def mapDouble(z: Double): Int = map(d2i(z))
 
   def mapColors(f: Int => Int): ColorMap =
     new IntColorMap(breaksToColors.map { case (key, color) => (key, f(color)) }, options)
 
   def mapColorsToIndex(): ColorMap =
     new IntColorMap(orderedBreaks.zipWithIndex.toMap, options)
 
-  def withNoDataColor(color: Int): ColorMap =
-    new IntColorMap(breaksToColors, options.copy(noDataColor = color))
+  def withNoDataColor(color: Int): ColorMap = {
+    new IntColorMap(
+      breaksToColors,
+      new Options(new MapStrategy(
+        options.strategy.boundary,
+        color,
+        options.strategy.fallbackValue,
+        options.strategy.strict
+      ))
+    )
+  }
```

Comment:
These are ugly. It would be much cleaner with `MapStrategy.copy`, except that would require it to be a `case class`. Case classes generate more bytecode, but if their instantiation cost isn't terrible, it might be worth it to reduce code smell here.